storagenode/contact/service.go

```diff
@@ -20,7 +20,7 @@ var mon = monkit.Package()
 // Service is the contact service between storage nodes and satellites
 type Service struct {
 	log       *zap.Logger
-	self      *overlay.NodeDossier
+	self      overlay.NodeDossier
```

Comment:
just out of curiosity why make this a value instead of a pointer?

Reply:
Solely for the sake of being able to pass the result of `peer.Kademlia.RoutingTable.Local()` into the `NewService` constructor. But also, it looks like that's how we generally pass these around. We don't _need_ it to be a pointer (it doesn't need to be changed from elsewhere), and the overhead of copying the extra bytes for copy-by-value is probably outweighed by the overhead of gc stuff when we use a pointer.

Reply:
I noticed that too... I bet it's so this service doesn't accidentally overwrite any of the nodeDossier data